projects/driver_controls/test/test_gpio_expander.c

```diff
@@ -0,0 +1,55 @@
+#include "unity.h"
+#include "log.h"
+#include "test_helpers.h"
+#include "gpio_expander.h"
+#include "i2c.h"
+
+void prv_callback(GPIOExpanderPin pin, GPIOState state, void *context) { }
+
+void setup_test(void) {
+  // GPIO initialization
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+
+  I2CSettings settings = {
+    .speed = I2C_SPEED_FAST,
+    .sda = { GPIO_PORT_B, 9 },
+    .scl = { GPIO_PORT_B, 8 }
+  };
+
+  i2c_init(I2C_PORT_1, &settings);
+
+  GPIOAddress address = { GPIO_PORT_A, 0 };
+  TEST_ASSERT_OK(gpio_expander_init(address, I2C_PORT_1));
+}
+
+void teardown_test(void) { }
+
+void test_gpio_expander_init_pin(void) {
```

Comment:
Reminder that tests should be considered isolated and the idea with `setup_test` is that everything is reset.